src/kdtree.rs

```diff
@@ -288,6 +288,44 @@ impl<A: Float + Zero + One, T, U: AsRef<[A]>> KdTree<A, T, U> {
         }
     }
 
+    pub fn remove(&mut self, point: &U, data: &T) -> Result<(), ErrorKind> {
+        if let Err(err) = self.check_point(point.as_ref()) {
+            return Err(err);
+        }
+        if let (Some(mut points), Some(mut bucket)) = (self.points.take(), self.bucket.take()) {
+            let point_pos = points.iter().position(|x| x == point);
```

Comment:
we could iterate by index to verify both point and data match
also current implementation short-circuits(`position`), which doesn't seem to handle the scenario with more than one match